tally/snapshot.go

```diff
@@ -102,6 +114,9 @@ func (snapshot *Snapshot) Aggregate(child *Snapshot) {
 	for key, timings := range child.timings {
 		snapshot.timings[key] = append(snapshot.timings[key], timings...)
 	}
+	for key, value := range child.gauges {
+		snapshot.Gauge(key, value, false)
```

Comment:
I'm not sure this is actually right. We don't keep track of the order of updates, so it gets really messy to have the child processes feeding into this central state here. I'm not actually sure what the best solution is :/